.pre-commit-config.yaml

```
@@ -23,8 +23,11 @@ repos:
       - id: mixed-line-ending
       - id: name-tests-test
         args: ["--pytest-test-first"]
+      - id: no-commit-to-branch #default is master and main
+        name: don't commit to main branch. If this pre-commit fails, rename `git branch -m <newname>` then rebase `git rebase -i upstream\main`
```

```suggestion
        name: no commits to main branch. (to fix: rename `git branch -m <newname>`)
```
This shows up every time and adds extra linebreaks on my terminal when it is so long. I don't think we need to mention rebasing, this should just rename the branch and they should be good to go when committing again. A quick suggestion above.

---

Oops, it doesn't like the ":" in there. You could change to a semi-colon or remove it altogether?
I am not sure it even needs the suggestion for the fix because "no commits to main branch" is pretty explanatory and someone would just google how to rename I think? I definitely agree we don't want to go down the extra bash commands route!

---

Unfortunately, I think any url will either be too long, or if you put it through a shortener it would be too cryptic to be useful. I think the name just needs to be meaningful enough that someone could search what it means if they don't already know, we don't necessarily need to give the user an exact solution.

---

Instead of augmenting the description, is there an easy way to automate a comment on the PR if pre-commit-ci fails?  I'm guessing that people who are inexperienced enough to be developing against main will mostly not have installed pre-commit hooks locally, so the PR is the place where this will usually fail.

---

Using https://github.com/marketplace/actions/comment-pull-request I think this should be possible